chainer/functions/connection/convolution_2d.py

```diff
@@ -235,22 +230,36 @@ def _forward_grouped_convolution(self, x, W, b):
 
         xp = cuda.get_array_module(x)
 
-        _x = x.reshape(N, G, iCg, iH, iW)
+        _x = x.reshape((N, G, iCg, iH, iW))
         _x = xp.rollaxis(_x, 1)  # (G, N, iCg, iH, iW)
-        _W = W.reshape(G, oCg, iCg, kH, kW)
+
+        _W = W.reshape((G, oCg, iCg, kH, kW))
         if b is not None:
-            _b = b.reshape(G, oCg)
+            _b = b.reshape((G, oCg))
+
+        if self._use_ideep:
+            _ys = intel64.ideep.mdarrayVector()
+        else:
+            _ys = []
 
-        _ys = []
         for g in six.moves.range(G):
             _bg = None if b is None else _b[g, ]
             if xp is numpy:
                 _y, = self._forward_cpu_core(_x[g, ], _W[g, ], _bg)
             else:
                 _y, = self._forward_gpu_core(_x[g, ], _W[g, ], _bg)
-            _ys.append(_y)
 
-        y = xp.concatenate(_ys, axis=1)  # (N, oC, oH, oW)
+            if self._use_ideep:
+                _ys.push_back(_y)
+            else:
+                _ys.append(_y)
+
+        # (N, oC, oH, oW)
+        if self._use_ideep:
+            y = intel64.ideep._ideep4py.concat.Forward(_ys, 1)
```

Comment:
Instead of branching every time an element is pushed to `_ys`, how about constructing `_ys` as a list and converting to a `mdarrayVector` here?